public/js/actions/AtomActions/getSuggestionsForLatestContent.js

```diff
@@ -0,0 +1,148 @@
+import {getTagsForContent} from '../../services/capi';
+import {mostViewed} from '../../services/ophan';
+import {fetchTargetsForTag} from '../../services/TargetingApi';
+import AtomsApi from '../../services/AtomsApi';
+import {atomPropType} from '../../constants/atomPropType.js';
+import {logError} from '../../util/logger';
+import {PropTypes} from 'react';
+
+function requestSuggestionsForLatestContent() {
+  return {
+    type:       'SUGGESTIONS_FOR_LATEST_CONTENT_REQUEST',
+    receivedAt: Date.now()
+  };
+}
+
+function receiveSuggestionsForLatestContent(suggestionsForLatestContent) {
+  return {
+    type:       'SUGGESTIONS_FOR_LATEST_CONTENT_RECEIVE',
+    suggestionsForLatestContent,
+    receivedAt: Date.now()
+  };
+}
+
+function errorReceivingSuggestionsForLatestContent(error) {
+  logError(error);
+  return {
+    type:       'SHOW_ERROR',
+    message:    'Could not get suggests for latest content',
+    error:      error,
+    receivedAt: Date.now()
+  };
+}
+
+const distinct = array => array.filter((value, index, self) => self.indexOf(value) === index);
+const flatten = array => [].concat.apply([], array);
+
+//Returns set of all tags in the given content
+function getTags(contentArray) {
+  return distinct(flatten(
+    contentArray.map(content => content.tags.filter(tag => tag.type === "keyword").map(tag => tag.id))
+  ));
+}
+
+//Returns a map of tagId to suggestions from the targeting api, filtering out non-atoms
+function getTagToTargetUrls(tags) {
+  const tagToTargetAtoms = {};
+
+  const fetch = (tag) => {
+    return fetchTargetsForTag(tag)
+      .then(targets => {
+        const filtered = targets.filter(target => target.url.includes("/atom/"));
+        const urls = filtered.map(target => target.url);
+        if (filtered.length > 0) {
+          tagToTargetAtoms[tag] = (tagToTargetAtoms[tag]) ? tagToTargetAtoms[tag].tag(urls) : urls;
```

Comment:
Regardless of which style you prefer, I don't think the consequent is every going to be called here. As each tag is unique, you can just write `tagToTargetAtoms[tag] = urls`.

Reply:
good point, I'm really not sure why I wrote that...